controllers/argocd/route_test.go

```diff
@@ -213,6 +213,123 @@ func TestReconcileRouteUnsetsInsecure(t *testing.T) {
 	}
 }
 
+func TestReconcileRouteApplicationSet(t *testing.T) {
+	routeAPIFound = true
+	ctx := context.Background()
+	logf.SetLogger(ZapLogger(true))
+	argoCD := makeArgoCD(func(a *argoproj.ArgoCD) {
+
+		a.Spec.ApplicationSet = &argoproj.ArgoCDApplicationSet{
+			WebhookServer: argoproj.WebhookServerSpec{
+				Host: "webhook-test.org",
+				Route: argoproj.ArgoCDRouteSpec{
+					Enabled: true,
+					Labels:  map[string]string{"my-key": "my-value"},
+				},
+			},
+		}
+	})
+
+	resObjs := []client.Object{argoCD}
+	subresObjs := []client.Object{argoCD}
+	runtimeObjs := []runtime.Object{}
+	sch := makeTestReconcilerScheme(argoproj.AddToScheme, configv1.Install, routev1.Install)
+	cl := makeTestReconcilerClient(sch, resObjs, subresObjs, runtimeObjs)
+	r := makeTestReconciler(cl, sch)
+
+	assert.NoError(t, createNamespace(r, argoCD.Namespace, ""))
+
+	req := reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      testArgoCDName,
+			Namespace: testNamespace,
+		},
+	}
+
+	_, err := r.Reconcile(context.TODO(), req)
+	assert.NoError(t, err)
+
+	loaded := &routev1.Route{}
+	err = r.Client.Get(ctx, types.NamespacedName{Name: testArgoCDName + "-applicationset-controller-webhook", Namespace: testNamespace}, loaded)
+	fatalIfError(t, err, "failed to load route %q: %s", testArgoCDName+"-server", err)
+
+	if diff := cmp.Diff("my-value", loaded.Labels["my-key"]); diff != "" {
+		t.Fatalf("failed to reconcile route:\n%s", diff)
+	}
+
+	wantTLSConfig := &routev1.TLSConfig{
+		InsecureEdgeTerminationPolicy: routev1.InsecureEdgeTerminationPolicyRedirect,
+		Termination:                   routev1.TLSTerminationEdge,
+	}
+	if diff := cmp.Diff(wantTLSConfig, loaded.Spec.TLS); diff != "" {
+		t.Fatalf("failed to reconcile route:\n%s", diff)
+	}
+
+	if diff := cmp.Diff(argoCD.Spec.ApplicationSet.WebhookServer.Host, loaded.Spec.Host); diff != "" {
+		t.Fatalf("failed to reconcile route:\n%s", diff)
+	}
+}
+
+func TestReconcileRouteApplicationSetTls(t *testing.T) {
+	routeAPIFound = true
+	ctx := context.Background()
+	logf.SetLogger(ZapLogger(true))
+	argoCD := makeArgoCD(func(a *argoproj.ArgoCD) {
+
+		a.Spec.ApplicationSet = &argoproj.ArgoCDApplicationSet{
+			WebhookServer: argoproj.WebhookServerSpec{
+				Host: "webhook-test.org",
+				Route: argoproj.ArgoCDRouteSpec{
+					Enabled: true,
+					TLS: &routev1.TLSConfig{
+						Termination: "passthrough",
+					},
```

Comment:
Do we want to test the other values in TLSConfig as well (key, ca cert, etc), since these were reported as part of the original bug?

Reply:
:+1: